modules/gfac/gfac-impl/src/main/java/org/apache/airavata/gfac/impl/task/SCPDataStageTask.java

```diff
@@ -111,24 +136,162 @@ public TaskStatus execute(TaskContext taskContext) {
                         status.setReason("File name is null");
                     }
                     return status;
+                } else {
+                    return transferFiles(processContext, subTaskModel, taskContext, processState);
                 }
+
             } else {
                 status.setState(TaskState.FAILED);
                 status.setReason("Invalid task invocation, Support " + ProcessState.INPUT_DATA_STAGING.name() + " and " +
                         "" + ProcessState.OUTPUT_DATA_STAGING.name() + " process phases. found " + processState.name());
                 return status;
             }
 
+        } catch (TException e) {
+            String msg = "Couldn't create subTask model thrift model";
+            log.error(msg, e);
+            status.setState(TaskState.FAILED);
+            status.setReason(msg);
+            ErrorModel errorModel = new ErrorModel();
+            errorModel.setActualErrorMessage(e.getMessage());
+            errorModel.setUserFriendlyMessage(msg);
+            taskContext.getTaskModel().setTaskErrors(Collections.singletonList(errorModel));
+            return status;
+        }
+    }
+
+    private TaskStatus extractStringFromFile(DataStagingTaskModel subTaskModel, TaskContext taskContext, String arguments) {
+
+        TaskStatus status = new TaskStatus(TaskState.EXECUTING);
+
+        File tempFile = null;
+        try {
+            if (arguments != null && !arguments.isEmpty()) {
+
+                String[] allArgs = arguments.split(",");
+                tempFile = File.createTempFile("temp-output", ".tmp");
+                URI sourceURI = new URI(subTaskModel.getSource());
+
+                log.info("Downloading file " + sourceURI.getPath() + " to temporary file " + tempFile);
+
+                taskContext.getParentProcessContext().getJobSubmissionRemoteCluster().copyFrom(sourceURI.getPath(), tempFile.getPath());
+
+                StringBuilder result = new StringBuilder();
+                // this is to identify that output is parsed
+                result.append("parsed-out: ");
+
+                // if the file size is grater than 2 MB, skip parsing to avoid possible OOM issues
```

Comment:
I don't think the OOM issue is a concern. The code isn't reading the entire file into memory. The BufferedReader only uses a relatively small buffer for reading the file.

One file size concern though is that if it is very large it may fill up the disk. If I'm understanding the code correctly this transfers the file from the compute resource to a temp file (in /tmp?) on the gfac server.  We generally don't provision a lot of storage space on gfac so /tmp may only have a few GBs, but some output files can be very large.  That would be a misconfiguration, but it would also disrupt service.

However, maybe not worth trying to solve at this time. I guess you could run something over SSH to check the size of the file before transferring it.

Reply:
Agreed. I misjudged OOM issue here. Thanks for pointing that out. Will remove this check and do a size check before transferring the file.